R/zchunk_L241.en_newtech_nonco2.R

```diff
@@ -1,107 +1,286 @@
+    # "Ensure only regions that have first gen biofuels get read in"
+    # (This is where the old data system had the note - TODO: better way to handle this,
+    # probably these technologies should pull from historical data. I opened an issue about it.)
+
+    # Use the may be historic new technology emission coefficients to replace the emission
+    # coefficients in the min model base year by a supplysecotr, subsector, stub.technology identifier
+    # called sector_tech_id.
+    A41.tech_coeff %>%
+      select("supplysector", "subsector", "stub.technology", "may.be.historic") %>%
+      na.omit %>%
+      unite(sector_tech_id, c("supplysector", "subsector", "stub.technology"), remove = FALSE) ->
+      L241.maybe_historic
+
+    # Replace the new technology emission coefficients in the model base year with the may be historic
+    # values from the technology assumption file by the unique supplysector, subsector,
+    # stub.technology identifier.
+    L241.nonco2_tech_coeff %>%
+      unite(sector_tech_id, c("supplysector", "subsector", "stub.technology"), remove = FALSE) %>%
+      mutate(year = replace(year, sector_tech_id %in% L241.maybe_historic$sector_tech_id, min(BASE_YEARS))) %>%
+      select(-sector_tech_id) ->
+      L241.nonco2_tech_coeff
+
+    # Replace the max emission reduction coefficients in the model base year with the may be historic
+    # values from the technology assumption file by the unique supplysector, subsector,
+    # stub.technology identifier.
+    L241.nonco2_max_reduction %>%
+      unite(sector_tech_id, c("supplysector", "subsector" , "stub.technology"), remove = FALSE) %>%
+      mutate(year = replace(year, sector_tech_id %in% L241.maybe_historic$sector_tech_id, min(BASE_YEARS))) %>%
+      select(-sector_tech_id) ->
+      L241.nonco2_max_reduction
+
+    # Replace the steepness of emission reduction coefficients in the model base year with the
+    # may be historic values from the technology assumption file by the unique supplysecotr, subsector,
+    # stub.technology identifier.
+    L241.nonco2_steepness %>%
+      unite(sector_tech_id, c("supplysector", "subsector" , "stub.technology"), remove = FALSE) %>%
+      mutate(year = replace(year, sector_tech_id %in% L241.maybe_historic$sector_tech_id, min(BASE_YEARS))) %>%
+      select(-sector_tech_id) ->
+      L241.nonco2_steepness
+
+
+    # Create a vector of the first generation biofuel technologies this vector will be used to
+    # remove non-applicable first-genbio technologies from the data frames.
+    L241.firstgenbio_techs <- c("corn ethanol", "sugarbeet ethanol", "sugar cane ethanol", "biodiesel")
+
+    # Create region ethanol type and region biodiesel type identifiers from the
+    # region-specific assumptions in the energy system. These identifiers will be
+    # used to remove non-applicable first-genbio technologies.
+    En_A_regions %>%
+      unite(col = region_eth, region, ethanol, sep = "~", remove = FALSE) %>%
```

Comment:
Um I always use the `~` as separator if I have to do a unite and separate. In the past I have run into issues when one of the variable being united/separated contains as space , `-` , or `_` already but I have yet to run into `~` existing in a variable name.

 If you want I can use something else.